### PR TITLE
update MALT report to use ES

### PR DESCRIPTION
**Pull request by czue**
This updates the MALT to use ES instead of sofabed.

I took a long and roundabout road but am happy where it ended up. Definitely easier to review :post_office: . You can ignore the commit comments.

@dimagi/scale-team and @sravfeyn probably best to review. cc @calellowitz @millerdev

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/data_analytics/tests/test_analytics.py, line 28 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 31 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 33 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 36 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 38 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 41 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 43 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 46 - E121 continuation line under-indented for hanging indent

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/data_analytics/tests/test_analytics.py, line 28 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 31 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 33 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 36 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 38 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 41 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 43 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 46 - E121 continuation line under-indented for hanging indent

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/data_analytics/tests/test_analytics.py, line 28 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 31 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 33 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 36 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 38 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 41 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 43 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_analytics.py, line 46 - E121 continuation line under-indented for hanging indent

**Comment by czue** [comment]
ok I think this is ready now

File: corehq/apps/data_analytics/analytics.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from corehq.apps.es.aggregations import AggregationTerm, NestedAggregationHelper
```
Comment (benrudolph) [inline corehq/apps/data_analytics/analytics.py]: are we going with `analytics.py` as the convention or `esaccessors.py`? i see both in the repo. some of the analytics files have couch calls though

File: corehq/apps/data_analytics/tests/test_analytics.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from corehq.pillows.xform import XFormPillow
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import generate_cases
+from dimagi.utils.dates import DateSpan
+from pillowtop.es_utils import completely_initialize_pillow_index
+
+
+class MaltAnalyticsTest(TestCase):
+    dependent_apps = [
+        'corehq.apps.sofabed'
```
Comment (benrudolph) [inline corehq/apps/data_analytics/tests/test_analytics.py]: isn't this removed?

Comment (benrudolph) [inline corehq/apps/data_analytics/tests/test_analytics.py]: nvm i see

File: corehq/apps/es/aggregations.py [hunk 3]
```diff
@@ -447,3 +447,42 @@ def __init__(self, name, datefield, interval, timezone=None):
...
         if timezone:
             self.body['time_zone'] = timezone
+
+
+AggregationTerm = namedtuple('AggregationTerm', ['name', 'field'])
+
+
+class NestedAggregationHelper(object):
+    """
+    Helepr to run nested term-based queries (equivalent to SQL group-by clauses)
```
Comment (benrudolph) [inline corehq/apps/es/aggregations.py]: helper

File: corehq/apps/es/aggregations.py [hunk 4]
```diff
@@ -447,3 +447,42 @@ def __init__(self, name, datefield, interval, timezone=None):
 
         if timezone:
             self.body['time_zone'] = timezone
+
+
+AggregationTerm = namedtuple('AggregationTerm', ['name', 'field'])
+
+
+class NestedAggregationHelper(object):
```
Comment (benrudolph) [inline corehq/apps/es/aggregations.py]: slick

File: corehq/form_processor/utils/xform.py [hunk 5]
```diff
@@ -34,7 +34,9 @@ class TestFormMetadata(jsonobject.JsonObject):
     xmlns = jsonobject.StringProperty(default='http://openrosa.org/formdesigner/form-processor')
     app_id = jsonobject.StringProperty(default='123')
     form_name = jsonobject.StringProperty(default='New Form')
+    device_id = jsonobject.StringProperty(default='DEV IL')
```
Comment (benrudolph) [inline corehq/form_processor/utils/xform.py]: :imp:

**Comment by benrudolph** [comment]
:+1: on :white_check_mark:

File: corehq/apps/data_analytics/analytics.py (hunk 1 above)
Comment (czue) [inline corehq/apps/data_analytics/analytics.py]: good point. i think esaccessors is good if it's all es. changed.

File: corehq/apps/data_analytics/tests/test_analytics.py (hunk 2 above)
Comment (czue) [inline corehq/apps/data_analytics/tests/test_analytics.py]: (removed)

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/data_analytics/tests/test_esaccessors.py, line 25 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 28 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 30 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 33 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 35 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 38 - E121 continuation line under-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 40 - E126 continuation line over-indented for hanging indent
* corehq/apps/data_analytics/tests/test_esaccessors.py, line 43 - E121 continuation line under-indented for hanging indent

**Comment by czue** [comment]
alrighty responded to feedback and fixed an issue. probably easiest to just review the last few commits individually

**Comment by benrudolph** [comment]
:+1:

**Pull request merged**